### PR TITLE
Fix API getting stuck in an endless loop under a certain unauthorized scenario

### PR DESCRIPTION
**Pull request by peppy**
While testing older versions of lazer I came across a case where the server would return an "Unauthorized" message but will not necessarily trigger the associated web exception. This aims to cover that case, while also adding extra protection against getting stuck in loops.

---

Resolves a case where the game could get in a state where logging out was not possible yet the client would never connect.

### CODE REVIEW

File: osu.Game/Online/API/APIAccess.cs [hunk 1]
```diff
@@ -148,7 +148,7 @@ private void run()
                         // The Success callback event is fired on the main thread, so we should wait for that to run before proceeding.
                         // Without this, we will end up circulating this Connecting loop multiple times and queueing up many web requests
                         // before actually going online.
-                        while (State != APIState.Online)
+                        while (State > APIState.Offline)
```
Comment (smoogipoo) [inline osu.Game/Online/API/APIAccess.cs]: This never exits, causing the API to never handle requests.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Online/API/APIAccess.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Online/API/APIAccess.cs]: under what circumstances?

Comment (smoogipoo) [reply osu.Game/Online/API/APIAccess.cs]: After the api becomes `Online`. Basically no requests besides the user one (above) happen.

**Review by smoogipoo — APPROVED** [review]

File: osu.Game/Online/API/APIAccess.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Online/API/APIAccess.cs]: oops

**Pull request merged**